Sources/App/entrypoint.swift

```diff
@@ -1,21 +1,29 @@
 import Vapor
 import Logging
+import NIOCore
+import NIOPosix
 
 @main
 enum Entrypoint {
     static func main() async throws {
         var env = try Environment.detect()
         try LoggingSystem.bootstrap(from: &env)
         
-        let app = Application(env)
-        defer { app.shutdown() }
+        let app = try await Application.make(env)
+
+        let executorTakeoverSuccess = NIOSingletons.unsafeTryInstallSingletonPosixEventLoopGroupAsConcurrencyGlobalExecutor()
```

Comment:
I wonder if we should include a comment here to reassure people who worry about the use of an `unsafe` API.

Reply:
Or maybe you want to "vaporise" it and give it a friendlier name? Potentially dependent on some env var?

Another idea would be Vapor owning the main entrypoint and calling that for you at the right point in time (depending on some env var or argument or so). Currently (we may be able to relax that -- see below), the method needs to be called _on the main thread_ which will only happen if the call either happens in a synchronous main function or an an asynchronous main function before the first `await`. So the code looks good as is (because it's the first line) but if somebody put a `try? await Task.sleep(nanoseconds: 123)` in front of it, it would likely just return `false`.

[I opted for the 'only from main thread' restriction because it would be quite something to allow users to replace the default concurrency executor whilst running on the previous default concurrency executor... That is not to say that we couldn't allow that but we should chat to some Swift runtime folks first to get their opinion).]

Reply:
I'll look at internalising it when I look at the service lifecycle